code/datums/components/label.dm

```diff
@@ -0,0 +1,87 @@
+
+/datum/component/label/Initialize(_label_name)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	label_name = _label_name
+	apply_label()
+
+/datum/component/label/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/OnAttackby)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/Examine)
+
+/datum/component/label/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_PARENT_ATTACKBY, COMSIG_PARENT_EXAMINE))
+
+/**
+	This proc will fire after the parent is hit by a hand labeler which is trying to apply another label.
+	Since the parent already has a label, it will remove the old one from the parent's name, and apply the new one.
+*/
+/datum/component/label/InheritComponent(datum/component/label/new_comp , i_am_original, list/arguments)
+	remove_label()
+	if(new_comp)
+		label_name = new_comp.label_name
+	else
+		label_name = arguments[1]
+	apply_label()
+
+/**
+	This proc will trigger when any object is used to attack the parent.
+
+	If the attacking object is not a hand labeler, it will return.
+	If the attacking object is a hand labeler it will restore the name of the parent to what it was before this component was added to it, and the component will be deleted.
+
+	Arguments:
+	* source: The parent.
+	* attacker: The object that is hitting the parent.
+	* user: The mob who is wielding the attacking object.
+*/
+/datum/component/label/proc/OnAttackby(datum/source, obj/item/attacker, mob/user)
+	// If the attacking object is not a hand labeler or its mode is 1 (has a label ready to apply), return.
+	// The hand labeler should be off (mode is 0), in order to remove a label.
+	var/obj/item/hand_labeler/labeler = attacker
+	if(!istype(labeler) || labeler.mode)
+		return
+
+	remove_label()
+	playsound(parent, 'sound/items/poster_ripped.ogg', 20, TRUE)
+	to_chat(user, "<span class='warning'>You remove the label from [parent].</span>")
+	qdel(src) // Remove the component from the object.
+
+/**
+	This proc will trigger when someone examines the parent.
+	It will attach the text found in the body of the proc to the `examine_list` and display it to the player examining the parent.
+
+	Arguments:
+	* source: The parent.
+	* user: The mob exmaining the parent.
+	* examine_list: The current list of text getting passed from the parent's normal examine() proc.
+*/
+/datum/component/label/proc/Examine(datum/source, mob/user, list/examine_list)
+	examine_list += "<span class='notice'>It has a label with some words written on it. Use a hand labeler to remove it.</span>"
+
+/// Applies a label to the name of the parent in the format of: "parent_name (label)"
+/datum/component/label/proc/apply_label()
+	var/atom/owner = parent
+	owner.name += " ([label_name])"
+
+/// Removes the label from the parent's name
+/datum/component/label/proc/remove_label()
+	var/atom/owner = parent
+	owner.name = replacetext(owner.name, "([label_name])", "") // Remove the label text from the parent's name, wherever it's located.
```

Comment:
Wouldn't it be best to look for `" ([label_name])"` with the space? If another effect puts text behind the name then there will be two spaces in between if the label was applied first. Can't quickly think of an effect that does such thing.

Reply:
Better to just cover all possible situations and future proof it I say :P